### PR TITLE
Yang model support for SRV6 CRM

### PR DESCRIPTION
**Pull request by kperumalbfn**
#### Why I did it
To support Yang models for SRV6 CM

#### How I did it
Added yang models for SRV6 MY_SID_ENTRY and Nexthop

#### How to verify it
Added SRV6 CRM yang tests.

#### Which release branch to backport (provide reason below if selected)
202111

### CODE REVIEW

**Comment by zhangyanzhao** [comment]
blocks #9383  merge

**Comment by kperumalbfn** [comment]
@prsunny @qiluo-msft Could you please review and merge this.

**Comment by dgsudharsan** [comment]
@qiluo-msft @zhangyanzhao This is a very import PR as it resolves break in yang. Can we expedite?

root@r-panther-13:~# config interface breakout Ethernet0 4x10G[25G]
Do you want to Breakout the port, continue? [y/N]: y

Running Breakout Mode : 1x100G[50G,40G,25G,10G]
Target Breakout Mode : 4x10G[25G]

Ports to be deleted :
 {
    "Ethernet0": "100000"
}
Ports to be added :
 {
    "Ethernet0": "10000",
    "Ethernet1": "10000",
    "Ethernet2": "10000",
    "Ethernet3": "10000"
}

After running Logic to limit the impact

Final list of ports to be deleted :
 {
    "Ethernet0": "100000"
}
Final list of ports to be added :
 {
    "Ethernet0": "10000",
    "Ethernet1": "10000",
    "Ethernet2": "10000",
    "Ethernet3": "10000"
}
Note: Below table(s) have no YANG models:
KDUMP, SNMP, SNMP_COMMUNITY, WJH, WJH_CHANNEL, XCVRD_LOG,
sonic_yang(3):All Keys are not parsed in CRM
dict_keys(['srv6_my_sid_entry_high_threshold', 'srv6_my_sid_entry_low_threshold', 'srv6_my_sid_entry_threshold_type', 'srv6_nexthop_high_threshold', 'srv6_nexthop_low_threshold', 'srv6_nexthop_threshold_type'])
sonic_yang(3):exceptionList:[]
sonic_yang(3):Data Loading Failed:All Keys are not parsed in CRM
dict_keys(['srv6_my_sid_entry_high_threshold', 'srv6_my_sid_entry_low_threshold', 'srv6_my_sid_entry_threshold_type', 'srv6_nexthop_high_threshold', 'srv6_nexthop_low_threshold', 'srv6_nexthop_threshold_type'])
Data Loading Failed
All Keys are not parsed in CRM
dict_keys(['srv6_my_sid_entry_high_threshold', 'srv6_my_sid_entry_low_threshold', 'srv6_my_sid_entry_threshold_type', 'srv6_nexthop_high_threshold', 'srv6_nexthop_low_threshold', 'srv6_nexthop_threshold_type'])
ConfigMgmt Class creation failed
Failed to break out Port. Error: Failed to load the config. Error: ConfigMgmtDPB Class creation failed

File: src/sonic-yang-models/tests/yang_model_tests/tests/crm.json [hunk 1]
```diff
@@ -115,5 +115,43 @@
     },
     "MPLS_NH_WITH_CORRECT_USED_VALUE": {
 	"desc": "MPLS_NH_WITH_CORRECT_USED_VALUE no failure."
+    },
+    "SRV6_MY_SID_ENTRY_WITH_WRONG_THRESHOLD_TYPE": {
```
Comment (qiluo-msft) [inline src/sonic-yang-models/tests/yang_model_tests/tests/crm.json:119]: >     [](http://example.com/codeflow?start=0&length=4)

This file is mixing tabs and spaces. Could you reformat the whole file?

File: src/sonic-yang-models/yang-models/sonic-crm.yang [hunk 2]
```diff
@@ -404,6 +404,58 @@ module sonic-crm {
                     type threshold;
                 }
 
+                leaf srv6_my_sid_entry_threshold_type {
+                    description "CRM threshold support for SRV6 my_sid entries";
+
+                    must "(((current()='PERCENTAGE' or current()='percentage') and
+                        ../srv6_my_sid_entry_high_threshold<100 and
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-crm.yang:411]: >< [](http://example.com/codeflow?start=59&length=1)

`<=` ?

**Comment by qiluo-msft** [comment]
@kperumalbfn This is yang model PR related to new fields in "srv6_my_sid_entry", "srv6_nexthop"

**Comment by kperumalbfn** [comment]
Yes @qiluo-msft. This is model for SRV6 CRM. Will address your comments and update the PR.

**Comment by zhangyanzhao** [comment]
Need SRv6 feature experts to help on the review.

**Comment by zhangyanzhao** [comment]
Need engage with SRv6 feature owner to check

**Comment by zhangyanzhao** [comment]
sent mail to Reshma

**Review by svshah-intel — APPROVED** [review]
Apart from "quilo-msft" comments, diffs look fine.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**